tests/test.rfb.js

```diff
@@ -2865,6 +2865,126 @@ describe('Remote Frame Buffer Protocol Client', function () {
             // error events do nothing
         });
     });
+
+    describe('Quality level setting', function () {
+        const defaultQuality = 6;
+
+        let client;
+
+        beforeEach(function () {
+            client = make_rfb();
+            sinon.spy(RFB.messages, "clientEncodings");
+        });
+
+        afterEach(function () {
+            RFB.messages.clientEncodings.restore();
+        });
+
+        it(`should equal ${defaultQuality} by default`, function () {
+            expect(client._qualityLevel).to.equal(defaultQuality);
+        });
+
+        it('should ignore non-integers when set', function () {
+            let newQuality;
+
+            newQuality = 3;
+            client.qualityLevel = newQuality;
+            expect(client._qualityLevel).to.equal(newQuality);
```

Comment:
Examining internal variables generally results in fragile tests, so we try to avoid that. Do you think you could examine the arguments to `RFB.messages.clientEncodings()` instead?

Reply:
Fixed